pkgs/build-support/rust/default.nix

```diff
@@ -239,7 +240,7 @@ stdenv.mkDerivation ((removeAttrs args ["depsExtraArgs"]) // lib.optionalAttrs u
   '';
 
   checkPhase = args.checkPhase or (let
-    argstr = "${lib.optionalString (checkType == "release") "--release"} --target ${target} --frozen";
+    argstr = "${lib.optionalString (checkType == "release") "--release"} --target ${target} --frozen ${builtins.concatStringsSep " " cargoTestFlags}";
```

Comment:
Doesn't this duplicate functionality of `checkFlags`/`checkFlagsArray`?

Reply:
i tried it, doesn't compile, since the checkFlags are behind the double dash

Reply:
> i tried it, doesn't compile, since the checkFlags are behind the double dash

I right, I forgot, that's for passing options to tests.

So, then this looks like a useful addition. However, could you rebase this on the `staging` branch? `buildRustPackage` was rewritten using hooks and is currently in `staging`/`staging-next`. Your addition should then be added to `pkgs/build-support/rust/hooks/cargo-check-hook.sh` and documented in the documentation for `cargoCheckHook`.

Reply:
BTW, it may be easier to redo this in staging rather than trying to rebase this PR. Feel free to close this PR and open a new one and ping me. Of course, if you prefer to rebase this PR, that's fine too!

Reply:
okay, i think i messed up, i checked out staging and applied the changes from scratch, afterwards i force pushed to my branch and changed the target branch in github... it seems like 12 additional people got tagged while doing so

Reply:
and i guess because of me messing up, ofborg might have added some labels that are not applicable, not sure tho...
sorry for that